rfcs/text/0014_scheduled_reports_new_plugin.md

```diff
@@ -0,0 +1,164 @@
+- Start Date: 2020-11-30
+- RFC PR: (leave this empty)
+- Kibana Issue: (leave this empty)
+
+# Summary
+
+With the ever-growing needs of Kibana to get data "out" to our users, so grows the responsibility of reporting and its current implementation. Historically, reporting was a "bolt-on" appliance custom built for specific pages, and wasn't a re-usable library allowing any page (or plugin) the ability to incorporate reports. Because of this approach, reporting doesn't scale well both for the owning team and its consumers, and this imbalance has grown to the point that it needs to be addressed.
+
+It's with these motivations in mind that reporting should now operate as a plugin, imported and used via platform APIs, so that Kibana pages and plugins can be reported-on in a more formal way.
+
+# Motivation
+
+At several points in the past, the team behind reporting has tried to formalize a mechanism that allows Kibana pages to be exported in a self-service fashion, but has had significant hardships in doing so. Part of these hardships are external to the team, but a lot of them are internal:
+
+- Numerous page-specific behaviors are embedded into our codebase.
+- No formal API on how to make pages themselves "exportable."
+- A convoluted REST API that doesn't lend itself well to being re-usable.
+- Requires at least two teams in order to integrate new pages.
+
+With the desire to build more sophisticated features, such as scheduled reports, the time has come to rethink how reporting operates so that page-specific behaviors belong to their respective owners. We see ownership start to become more concrete once we make the following statements:
+
+- Rendering characteristics of a page (Dashboards, for instance) should be maintained and owned by that team.
+- The general pipeline of reporting: queueing, export execution, browser binaries, persistance of assets, and the REST APIs are owned by the reporting team.
+- Other characteristics, such as licensing and role's, can be established in one (or both) places.
+
+At its core, we want to expand upon the features and fixes that can make reporting a killer-feature of Kibana. In order to do so, it's time to shed the maintenance cost and upkeep of application-specific logic to the underlying team's themselves.
+
+# Overall design
+
+The new plugin would follow Kibana's recommended and best practices: establishing a TypeScript-based plugin that others can add as a dependency and consume. The following is a brief overview of how an application would incorporate reporting into their application stack.
+
+## 1. Add a dependency on reporting
+In order to use reporting, a plugin simply has to add a dependency on reporting in their `kibana.json`:
```

Comment:
worth mentioning (even thou its visible from sample code) that it should be an optional dependency.
- importing something as optional will prevent you from importing any types from that plugin (or any other static assets)